app/models/postcode_local_restriction_search.rb

```diff
@@ -0,0 +1,53 @@
+class PostcodeLocalRestrictionSearch
+  UK_POSTCODE_PATTERN = %r{
+    \A
+    ([Gg][Ii][Rr] 0[Aa]{2}) # Unusual postcodes
+    |
+    (
+      # Outward code
+      (([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9][A-Za-z]?))))
+      \s?
+      [0-9][A-Za-z]{2} # Inward code
```

Comment:
I've added some info to the commit message now - I've also updated the comments to give an example of what outward and inward codes mean in postcode terms. 

The regex is from PostcodeService, I separated it out from being a big long block of characters so that it was readable.